diff -Nuard xmlrpc-c-0.9.9.orig/configure.in xmlrpc-c-0.9.9/configure.in
--- xmlrpc-c-0.9.9.orig/configure.in	2001-04-04 00:02:10.000000000 +0200
+++ xmlrpc-c-0.9.9/configure.in	2003-09-15 03:26:33.000000000 +0200
@@ -147,6 +147,7 @@
 
 AC_SUBST(AVAILABLE_MODULES)
 
+AM_CONDITIONAL(am__fastdepCXX, test yes = $enable_cplusplus)
 
 dnl =======================================================================
 dnl Checks for programs.
