--- xmlrpc-c-1.43.08/src/Makefile.orig	2016-07-22 04:51:19.000000000 +0200
+++ xmlrpc-c-1.43.08/src/Makefile	2024-10-27 07:56:23.799721028 +0100
@@ -15,7 +15,7 @@ ifeq ($(ENABLE_CPLUSPLUS),yes)
   SUBDIRS += cpp
 endif
 
-TARGET_LIBRARY_NAMES := libxmlrpc libxmlrpc_server
+TARGET_LIBRARY_NAMES := libxmlrpc-c libxmlrpc_server
 
 # We cannot build libxmlrpc_server_abyss if we can't also build Abyss,
 # because we need libxmlrpc_abyss in order to build the proper runtime
@@ -165,7 +165,7 @@ endif
 # shlibfn generates e.g. libxmlrpc.so.3.1
 # shliblefn generates e.g. libxmlrpc.so
 
-LIBXMLRPC = $(call shlibfn, libxmlrpc)
+LIBXMLRPC = $(call shlibfn, libxmlrpc-c)
 
 $(LIBXMLRPC): $(LIBXMLRPC_MODS:%=%.osh) \
   $(LIBXMLRPC_UTIL) \
@@ -180,10 +180,10 @@ LIBXMLRPC_SERVER = $(call shlibfn, libxm
 $(LIBXMLRPC_SERVER): \
   $(LIBXMLRPC_SERVER_MODS:%=%.osh) \
   $(LIBXMLRPC_UTIL) \
-  $(call shliblefn, libxmlrpc)
+  $(call shliblefn, libxmlrpc-c)
 $(LIBXMLRPC_SERVER): LIBOBJECTS = $(LIBXMLRPC_SERVER_MODS:%=%.osh)
 $(LIBXMLRPC_SERVER): LIBDEP = \
-   -L. -lxmlrpc $(XML_PARSER_LIBDEP) $(LIBXMLRPC_UTIL_LIBDEP)
+   -L. -lxmlrpc-c $(XML_PARSER_LIBDEP) $(LIBXMLRPC_UTIL_LIBDEP)
 
 LIBXMLRPC_SERVER_ABYSS = $(call shlibfn, libxmlrpc_server_abyss)
 
@@ -192,12 +192,12 @@ $(LIBXMLRPC_SERVER_ABYSS): \
   $(LIBXMLRPC_UTIL) \
   $(LIBXMLRPC_ABYSS) \
   $(call shliblefn, libxmlrpc_server) \
-  $(call shliblefn, libxmlrpc)
+  $(call shliblefn, libxmlrpc-c)
 $(LIBXMLRPC_SERVER_ABYSS): LIBOBJECTS = $(LIBXMLRPC_SERVER_ABYSS_MODS:%=%.osh)
 $(LIBXMLRPC_SERVER_ABYSS): LIBDEP = \
   -L. -lxmlrpc_server \
   -L$(LIBXMLRPC_ABYSS_DIR) -lxmlrpc_abyss \
-  -L. -lxmlrpc $(XML_PARSER_LIBDEP) $(LIBXMLRPC_UTIL_LIBDEP)
+  -L. -lxmlrpc-c $(XML_PARSER_LIBDEP) $(LIBXMLRPC_UTIL_LIBDEP)
 ifeq ($(MSVCRT),yes)
   $(LIBXMLRPC_SERVER_ABYSS):  LIBDEP += -lws2_32 -lwsock32
 endif
@@ -208,11 +208,11 @@ $(LIBXMLRPC_SERVER_CGI): \
   $(LIBXMLRPC_SERVER_CGI_MODS:%=%.osh) \
   $(LIBXMRPC_UTIL) \
   $(call shliblefn, libxmlrpc_server) \
-  $(call shliblefn, libxmlrpc)
+  $(call shliblefn, libxmlrpc-c)
 $(LIBXMLRPC_SERVER_CGI): LIBOBJECTS = $(LIBXMLRPC_SERVER_CGI_MODS:%=%.osh)
 $(LIBXMLRPC_SERVER_CGI): LIBDEP = \
   -L. -lxmlrpc_server \
-  -L. -lxmlrpc $(XML_PARSER_LIBDEP) $(LIBXMLRPC_UTIL_LIBDEP)
+  -L. -lxmlrpc-c $(XML_PARSER_LIBDEP) $(LIBXMLRPC_UTIL_LIBDEP)
 
 LIBXMLRPC_CLIENT = $(call shlibfn, libxmlrpc_client)
 
@@ -220,7 +220,7 @@ $(LIBXMLRPC_CLIENT): \
   $(LIBXMLRPC_CLIENT_MODS:%=%.osh) \
   $(TRANSPORT_MODS:%=%.osh) \
   $(LIBXMLRPC_UTIL) \
-  $(call shliblefn, libxmlrpc)
+  $(call shliblefn, libxmlrpc-c)
 $(LIBXMLRPC_CLIENT): LIBOBJECTS = \
   $(LIBXMLRPC_CLIENT_MODS:%=%.osh) \
   $(TRANSPORT_MODS:%=%.osh)
@@ -236,7 +236,7 @@ $(LIBXMLRPC_CLIENT): LIBOBJECTS = \
 
 LIBXMLRPC_CLIENT_LIBDEP = \
   -Lblddir/src -Lblddir/lib/libutil \
-  -lxmlrpc -lxmlrpc_util \
+  -lxmlrpc-c -lxmlrpc_util \
   $(XML_PARSER_LIBDEP) \
   $(TRANSPORT_LIBDEP) \
 
@@ -245,8 +245,8 @@ $(LIBXMLRPC_CLIENT): LIBDEP = \
   $(SOCKETLIBOPT) \
   $(THREAD_LIBS) \
 
-libxmlrpc.a: $(LIBXMLRPC_MODS:%=%.o)
-libxmlrpc.a: LIBOBJECTS = $(LIBXMLRPC_MODS:%=%.o)
+libxmlrpc-c.a: $(LIBXMLRPC_MODS:%=%.o)
+libxmlrpc-c.a: LIBOBJECTS = $(LIBXMLRPC_MODS:%=%.o)
 
 libxmlrpc_server.a: $(LIBXMLRPC_SERVER_MODS:%=%.o)
 libxmlrpc_server.a: LIBOBJECTS = $(LIBXMLRPC_SERVER_MODS:%=%.o)
--- xmlrpc-c-1.14.2/common.mk	2008-05-14 17:43:31.882708000 +0200
+++ xmlrpc-c-1.14.2/common.mk	2008-05-14 17:59:12.072278970 +0200
@@ -142,7 +142,7 @@
 ifneq ($(OMIT_XMLRPC_LIB_RULE),Y)
 
 LIBXMLRPC              = \
-  $(call shliblefn, $(BLDDIR)/src/libxmlrpc)
+  $(call shliblefn, $(BLDDIR)/src/libxmlrpc-c)
 LIBXMLRPC_CLIENT       = \
   $(call shliblefn, $(BLDDIR)/src/libxmlrpc_client)
 LIBXMLRPC_SERVER       = \
@@ -152,7 +152,7 @@
 LIBXMLRPC_SERVER_CGI   = \
   $(call shliblefn, $(BLDDIR)/src/libxmlrpc_server_cgi)
 
-LIBXMLRPC_A              = $(BLDDIR)/src/libxmlrpc.a
+LIBXMLRPC_A              = $(BLDDIR)/src/libxmlrpc-c.a
 LIBXMLRPC_CLIENT_A       = $(BLDDIR)/src/libxmlrpc_client.a
 LIBXMLRPC_SERVER_A       = $(BLDDIR)/src/libxmlrpc_server.a
 LIBXMLRPC_SERVER_ABYSS_A = $(BLDDIR)/src/libxmlrpc_server_abyss.a
--- xmlrpc-c-1.39.12/xmlrpc-c-config.main.orig	2018-02-18 18:00:53.710261758 +0100
+++ xmlrpc-c-1.39.12/xmlrpc-c-config.main	2018-02-18 21:02:28.870137355 +0100
@@ -46,7 +46,7 @@
 
 the_libdirs="-L$LIBINST_DIR $the_libdirs"
 
-the_libs="-lxmlrpc  ${LIBXML} -lxmlrpc_util -lpthread"
+the_libs="-lxmlrpc-c ${LIBXML} -lxmlrpc_util -lpthread"
 the_rpath="-R$LIBINST_DIR $the_rpath"
 the_wl_rpath="-Wl,-rpath,$LIBINST_DIR $the_wl_rpath"
 
--- xmlrpc-c-1.43.08/src/cpp/Makefile.orig	2016-07-22 04:51:19.000000000 +0200
+++ xmlrpc-c-1.43.08/src/cpp/Makefile	2024-10-27 08:18:45.757640890 +0100
@@ -130,7 +130,7 @@ $(LIBXMLRPC_CPP_SH): XmlRpcCpp.osh \
 
 $(LIBXMLRPC_CPP_SH): LIBOBJECTS = XmlRpcCpp.osh
 $(LIBXMLRPC_CPP_SH): LIBDEP = \
-  -Lblddir/src  -lxmlrpc_server -lxmlrpc \
+  -Lblddir/src  -lxmlrpc_server -lxmlrpc-c \
   $(XML_PARSER_LIBDEP) \
   $(LIBXMLRPC_UTIL_LIBDEP) \
 
@@ -143,7 +143,7 @@ $(LIBXMLRPCPP_SH): $(LIBXMLRPCPP_MODS:%=
 
 $(LIBXMLRPCPP_SH): LIBOBJECTS = $(LIBXMLRPCPP_MODS:%=%.osh)
 $(LIBXMLRPCPP_SH): LIBDEP = \
-  -Lblddir/src -lxmlrpc \
+  -Lblddir/src -lxmlrpc-c \
   $(XML_PARSER_LIBDEP) \
   -L$(LIBXMLRPC_UTILPP_DIR) -lxmlrpc_util++ \
   -L$(LIBXMLRPC_UTIL_DIR) -lxmlrpc_util \
@@ -161,7 +161,7 @@ $(LIBXMLRPC_SERVERPP_SH): $(LIBXMLRPC_SE
 $(LIBXMLRPC_SERVERPP_SH): LIBOBJECTS = $(LIBXMLRPC_SERVERPP_MODS:%=%.osh)
 $(LIBXMLRPC_SERVERPP_SH): LIBDEP = \
   -L. -l$(LIBXMLRPCPP_NAME) \
-  -Lblddir/src -lxmlrpc_server -lxmlrpc \
+  -Lblddir/src -lxmlrpc_server -lxmlrpc-c \
   $(XML_PARSER_LIBDEP) \
   -L$(LIBXMLRPC_UTILPP_DIR) -lxmlrpc_util++ \
   -L$(LIBXMLRPC_UTIL_DIR) -lxmlrpc_util \
@@ -182,7 +182,7 @@ $(LIBXMLRPC_SERVER_ABYSSPP_SH): $(LIBXML
 $(LIBXMLRPC_SERVER_ABYSSPP_SH): LIBOBJECTS = $(LIBXMLRPC_SERVER_ABYSSPP_MODS:%=%.osh)
 $(LIBXMLRPC_SERVER_ABYSSPP_SH): LIBDEP = \
   -L. -lxmlrpc_server++  -l$(LIBXMLRPCPP_NAME) \
-  -Lblddir/src -lxmlrpc_server_abyss -lxmlrpc_server -lxmlrpc \
+  -Lblddir/src -lxmlrpc_server_abyss -lxmlrpc_server -lxmlrpc-c \
   $(XML_PARSER_LIBDEP) \
   -L$(LIBXMLRPC_ABYSSPP_DIR) -lxmlrpc_abyss++ \
   -L$(LIBXMLRPC_ABYSS_DIR) -lxmlrpc_abyss \
@@ -203,7 +203,7 @@ $(LIBXMLRPC_SERVER_CGIPP_SH): $(LIBXMLRP
 $(LIBXMLRPC_SERVER_CGIPP_SH): LIBOBJECTS = $(LIBXMLRPC_SERVER_CGIPP_MODS:%=%.osh)
 $(LIBXMLRPC_SERVER_CGIPP_SH): LIBDEP = \
   -L. -lxmlrpc_server++  -l$(LIBXMLRPCPP_NAME) \
-  -Lblddir/src -lxmlrpc_server -lxmlrpc \
+  -Lblddir/src -lxmlrpc_server -lxmlrpc-c \
   $(XML_PARSER_LIBDEP) \
   -L$(LIBXMLRPC_UTILPP_DIR) -lxmlrpc_util++ \
   -L$(LIBXMLRPC_UTIL_DIR) -lxmlrpc_util \
@@ -222,7 +222,7 @@ $(LIBXMLRPC_SERVER_PSTREAMPP_SH): $(LIBX
 $(LIBXMLRPC_SERVER_PSTREAMPP_SH): LIBOBJECTS = $(LIBXMLRPC_SERVER_PSTREAMPP_MODS:%=%.osh)
 $(LIBXMLRPC_SERVER_PSTREAMPP_SH): LIBDEP = \
   -L. -lxmlrpc_server++ -l$(LIBXMLRPCPP_NAME) -lxmlrpc_packetsocket \
-  -Lblddir/src -lxmlrpc_server -lxmlrpc \
+  -Lblddir/src -lxmlrpc_server -lxmlrpc-c \
   $(XML_PARSER_LIBDEP) \
   -L$(LIBXMLRPC_UTILPP_DIR) -lxmlrpc_util++ \
   -L$(LIBXMLRPC_UTIL_DIR) -lxmlrpc_util \
@@ -244,7 +244,7 @@ $(LIBXMLRPC_CLIENTPP_SH): LIBOBJECTS = $
 $(LIBXMLRPC_CLIENTPP_SH): LIBDEP = \
   -L. -l$(LIBXMLRPCPP_NAME) -lxmlrpc_packetsocket \
   $(shell cat blddir/src/libxmlrpc_client.ldflags) \
-  -Lblddir/src -lxmlrpc_client -lxmlrpc \
+  -Lblddir/src -lxmlrpc_client -lxmlrpc-c \
   $(XML_PARSER_LIBDEP) \
   -L$(LIBXMLRPC_UTILPP_DIR) -lxmlrpc_util++ \
   -L$(LIBXMLRPC_UTIL_DIR) -lxmlrpc_util \
@@ -261,7 +261,7 @@ $(LIBXMLRPC_PACKETSOCKET_SH): $(LIBXMLRP
 $(LIBXMLRPC_PACKETSOCKET_SH): LIBOBJECTS = $(LIBXMLRPC_PACKETSOCKET_MODS:%=%.osh)
 $(LIBXMLRPC_PACKETSOCKET_SH): LIBDEP = \
   -L. -l$(LIBXMLRPCPP_NAME) \
-  -Lblddir/src -lxmlrpc \
+  -Lblddir/src -lxmlrpc-c \
   $(XML_PARSER_LIBDEP) \
   -L$(LIBXMLRPC_UTILPP_DIR) -lxmlrpc_util++ \
   -L$(LIBXMLRPC_UTIL_DIR) -lxmlrpc_util \
--- xmlrpc-c/src/xmlrpc.pc.cmake.orig	2013-03-03 17:06:35.285532023 +0100
+++ xmlrpc-c/src/xmlrpc.pc.cmake	2013-03-03 17:46:37.542097875 +0100
@@ -8,5 +8,5 @@
 Version:     @XMLRPC_C_VERSION@
 
 Requires.private:	@xmlrpc_pkgconfig_req@ xmlrpc_util
-Libs:			-L${libdir} -lxmlrpc @xmlrpc_pkgconfig_libs@
+Libs:			-L${libdir} -lxmlrpc-c @xmlrpc_pkgconfig_libs@
 Cflags:			-I${includedir}
--- xmlrpc-c-1.43.08/src/CMakeLists.txt.orig	2024-10-27 08:48:07.694762324 +0100
+++ xmlrpc-c-1.43.08/src/CMakeLists.txt	2024-10-27 08:49:39.967595773 +0100
@@ -47,8 +47,8 @@ add_subdirectory(cpp)
 
 
 
-### libxmlrpc.so
-add_library(xmlrpc SHARED
+### libxmlrpc-c.so
+add_library(xmlrpc-c SHARED
   base_global.c double.c parse_datetime.c parse_value.c resource.c trace.c version.c
   json.c
   ${xmlrpc_xml_parser}
@@ -57,19 +57,19 @@ add_library(xmlrpc SHARED
   xmlrpc_build.c xmlrpc_decompose.c xmlrpc_parse.c xmlrpc_serialize.c
   xmlrpc_authcookie.c)
 
-target_compile_options(xmlrpc
+target_compile_options(xmlrpc-c
   PRIVATE
   ${libxml_INCLUDES})
 
 ensc_pkgconfig(xmlrpc)
-target_link_libraries(xmlrpc ${libxml_LIBS} xmlrpc_util)
-list(APPEND lib_TARGETS xmlrpc)
+target_link_libraries(xmlrpc-c ${libxml_LIBS} xmlrpc_util)
+list(APPEND lib_TARGETS xmlrpc-c)
 
 ### libxmlrpc_client.so
 add_library(xmlrpc_client SHARED
   xmlrpc_client.c xmlrpc_client_global.c xmlrpc_server_info.c ${transport_SOURCES})
 
-target_link_libraries(xmlrpc_client xmlrpc ${client_LIBS})
+target_link_libraries(xmlrpc_client xmlrpc-c ${client_LIBS})
 set_target_properties(xmlrpc_client
   PROPERTIES
   COMPILE_FLAGS "${client_CFLAGS}")
@@ -79,7 +79,7 @@ ensc_pkgconfig(xmlrpc_client)
 ### libxmlrpc_server.so
 add_library(xmlrpc_server SHARED
   registry.c method.c system_method.c)
-target_link_libraries(xmlrpc_server xmlrpc)
+target_link_libraries(xmlrpc_server xmlrpc-c)
 list(APPEND lib_TARGETS xmlrpc_server)
 ensc_pkgconfig(xmlrpc_server)
 
--- xmlrpc-c-1.39.12/src/cpp/CMakeLists.txt.orig	2018-02-25 15:31:16.956791945 +0100
+++ xmlrpc-c-1.39.12/src/cpp/CMakeLists.txt	2018-02-25 15:37:15.140121190 +0100
@@ -4,13 +4,13 @@
 add_library(xmlrpc++ SHARED
   fault.cpp global.cpp
   outcome.cpp param_list.cpp value.cpp xml.cpp)
-target_link_libraries(xmlrpc++ xmlrpc_util++ xmlrpc pthread)
+target_link_libraries(xmlrpc++ xmlrpc_util++ xmlrpc-c pthread)
 list(APPEND lib_TARGETS xmlrpc++)
 ensc_pkgconfig(xmlrpc++)
 
 ####### libxmlrpc_cpp.so
 add_library(xmlrpc_cpp            SHARED XmlRpcCpp)
-target_link_libraries(xmlrpc_cpp xmlrpc xmlrpc_server)
+target_link_libraries(xmlrpc_cpp xmlrpc-c xmlrpc_server)
 list(APPEND lib_TARGETS xmlrpc_cpp)
 ensc_pkgconfig(xmlrpc_cpp)
 
--- xmlrpc-c/examples/CMakeLists.txt.orig	2011-03-20 17:08:15.832709568 +0100
+++ xmlrpc-c/examples/CMakeLists.txt	2011-03-20 17:12:17.396709568 +0100
@@ -21,7 +21,7 @@
 endif(ENABLE_CPLUSPLUS)
 
 
-set(base_LIBS xmlrpc)
+set(base_LIBS xmlrpc-c)
 ensc_add_example(json               c base)
 ensc_add_example(gen_sample_add_xml c base)
 
--- xmlrpc-c/tools/xml/CMakeLists.txt.orig	2013-03-03 17:51:15.572086205 +0100
+++ xmlrpc-c/tools/xml/CMakeLists.txt	2013-03-03 17:56:38.495405989 +0100
@@ -5,7 +5,7 @@
   ../lib/dumpvalue.c)
 
 target_link_libraries(xmlrpc_parsecall
-  xmlrpc
+  xmlrpc-c
   util)
 
 install(TARGETS xmlrpc_parsecall
