--- xmlrpc-c-1.43.08/common.mk.orig	2024-10-26 21:26:23.240673775 +0200
+++ xmlrpc-c-1.43.08/common.mk	2024-10-26 21:30:32.219324942 +0200
@@ -49,14 +49,14 @@ GCC_CXX_WARNINGS = $(GCC_WARNINGS)  -Wsy
 # -UNDEBUG (in any of various ways) to override this.
 #
 CFLAGS_COMMON = -DNDEBUG $(CFLAGS_PTHREAD)
-CXXFLAGS_COMMON = -DNDEBUG $(CFLAGS_PTHREAD)
+CXXFLAGS_COMMON = -DNDEBUG $(CFLAGS_PTHREAD) -fPIC
 
 ifeq ($(C_COMPILER_GNU),yes)
-  CFLAGS_COMMON += $(GCC_C_WARNINGS) -fno-common -g -O3
+  CFLAGS_COMMON += $(GCC_C_WARNINGS) -fno-common
 endif
 
 ifeq ($(CXX_COMPILER_GNU),yes)
-  CXXFLAGS_COMMON += $(GCC_CXX_WARNINGS) -g
+  CXXFLAGS_COMMON += $(GCC_CXX_WARNINGS)
 endif
 
 # -qrtti tell the IBM compilers to allow dynamic type casting.  Without it,
