--- xmlrpc-c-1.10.00.org/common.mk	2007-03-23 03:02:14.000000000 +0100
+++ xmlrpc-c-1.10.00/common.mk	2007-04-11 09:05:10.452171013 +0200
@@ -28,15 +28,15 @@
 # assertion and crash the program if it isn't really true.  You can add
 # -UNDEBUG (in any of various ways) to override this.
 #
-CFLAGS_COMMON = -DNDEBUG
+CFLAGS_COMMON = -DNDEBUG -fPIC
 CXXFLAGS_COMMON = -DNDEBUG
 
 ifeq ($(C_COMPILER_GNU),yes)
-  CFLAGS_COMMON += $(GCC_C_WARNINGS) -fno-common -g -O3
+  CFLAGS_COMMON += $(GCC_C_WARNINGS) -fno-common
 endif
 
 ifeq ($(CXX_COMPILER_GNU),yes)
-  CXXFLAGS_COMMON += $(GCC_CXX_WARNINGS) -g
+  CXXFLAGS_COMMON += $(GCC_CXX_WARNINGS)
 endif
 
 # -qrtti tell the IBM compilers to allow dynamic type casting.  Without it,
