--- xmlrpc-c/src/CMakeLists.txt.orig	2013-03-03 17:47:58.712094469 +0100
+++ xmlrpc-c/src/CMakeLists.txt	2013-03-03 17:50:33.195421319 +0100
@@ -22,7 +22,7 @@
 if(MUST_BUILD_LIBWWW_CLIENT)
   set(client_libwww_CFLAGS "${LIBWWW_CFLAGS}")
   set(client_libwww_LIBS   "${LIBWWW_LIBS}")
-  list(APPEND transport_SOURCES ${libwww_srcdir}/xmlrpc_libwww_transport.c ${libwww_srcdir}/xmlrpc_libwww_transport.h)
+  list(APPEND transport_SOURCES ${libwww_srcdir}/xmlrpc_libwww_transport.c)
 endif(MUST_BUILD_LIBWWW_CLIENT)
 
 if(MUST_BUILD_CURL_CLIENT)
