diff -ur xmlrpc-c-1.06.04/Makefile.common xmlrpc-c-1.06.04-cflags/Makefile.common
--- xmlrpc-c-1.06.04/Makefile.common	2006-09-04 14:47:37.000000000 +0200
+++ xmlrpc-c-1.06.04-cflags/Makefile.common	2006-09-04 14:48:59.000000000 +0200
@@ -28,15 +28,15 @@
 # assertion and crash the program if it isn't really true.  You can add
 # -UNDEBUG (in any of various ways) to override this.
 #
-CFLAGS_COMMON = -DNDEBUG
-CXXFLAGS_COMMON = -DNDEBUG
+CFLAGS_COMMON = -DNDEBUG $(OPTCFLAGS)
+CXXFLAGS_COMMON = -DNDEBUG $(OPTCXXFLAGS)
 
 ifeq ($(C_COMPILER_GNU),yes)
-  CFLAGS_COMMON += $(GCC_C_WARNINGS) -fno-common -g -O3
+  CFLAGS_COMMON += $(GCC_C_WARNINGS) -fno-common
 endif
 
 ifeq ($(CXX_COMPILER_GNU),yes)
-  CXXFLAGS_COMMON += $(GCC_CXX_WARNINGS) -g
+  CXXFLAGS_COMMON += $(GCC_CXX_WARNINGS) 
 endif
 
 DISTDIR = $(BUILDDIR)/$(PACKAGE)-$(VERSION)/$(SUBDIR)
diff -ur xmlrpc-c-1.06.04/lib/abyss/src/Makefile xmlrpc-c-1.06.04-cflags/lib/abyss/src/Makefile
--- xmlrpc-c-1.06.04/lib/abyss/src/Makefile	2006-09-04 14:47:37.000000000 +0200
+++ xmlrpc-c-1.06.04-cflags/lib/abyss/src/Makefile	2006-09-04 14:51:35.000000000 +0200
@@ -6,7 +6,7 @@
 include $(SRCDIR)/Makefile.config
 
 CFLAGS = $(CFLAGS_COMMON)
-CFLAGS += -D_UNIX
+CFLAGS += -D_UNIX -fPIC
 ifeq ($(ENABLE_ABYSS_THREADS),yes)
   THREAD_MODULE = thread_pthread
 else
diff -ur xmlrpc-c-1.06.04/lib/curl_transport/Makefile xmlrpc-c-1.06.04-cflags/lib/curl_transport/Makefile
--- xmlrpc-c-1.06.04/lib/curl_transport/Makefile	2005-09-19 03:20:14.000000000 +0200
+++ xmlrpc-c-1.06.04-cflags/lib/curl_transport/Makefile	2006-09-04 14:51:53.000000000 +0200
@@ -10,7 +10,7 @@
 # we need just the -I options for 'make dep'.  Plus, it's scary to think
 # of what any other compiler flag would do to our compile.
 
-CFLAGS = $(CFLAGS_COMMON) $(CFLAGS_PERSONAL) $(CADD)
+CFLAGS = $(CFLAGS_COMMON) $(CFLAGS_PERSONAL) $(CADD) -fPIC
 LDFLAGS = $(LADD)
 
 INCLUDES = -I$(SRCDIR) -I$(SRCDIR)/include -I$(SRCDIR)/lib/util/include \
@@ -48,4 +48,4 @@
 
 # Need this dependency for those who don't use Makefile.depend.
 # Without it, version.h doesn't get created.
-xmlrpc_curl_transport.lo: version.h
\ No newline at end of file
+xmlrpc_curl_transport.lo: version.h
diff -ur xmlrpc-c-1.06.04/lib/expat/xmlparse/Makefile xmlrpc-c-1.06.04-cflags/lib/expat/xmlparse/Makefile
--- xmlrpc-c-1.06.04/lib/expat/xmlparse/Makefile	2006-09-04 14:47:37.000000000 +0200
+++ xmlrpc-c-1.06.04-cflags/lib/expat/xmlparse/Makefile	2006-09-04 14:52:24.000000000 +0200
@@ -29,7 +29,7 @@
 libxmlrpc-c_xmlparse.la: $(LIBXMLRPC_XMLPARSE_OBJS)
 	$(LIBTOOL) --mode=link $(CCLD) -o $@ $(LIBLDFLAGS) $^
 
-CFLAGS = $(CFLAGS_COMMON) $(CFLAGS_PERSONAL) $(CADD)
+CFLAGS = $(CFLAGS_COMMON) $(CFLAGS_PERSONAL) $(CADD) -fPIC
 
 $(LIBXMLRPC_XMLPARSE_OBJS):%.lo:%.c
 	$(LIBTOOL) --mode=compile $(CC) -c $(INCLUDES) $(CFLAGS) $<
diff -ur xmlrpc-c-1.06.04/lib/expat/xmltok/Makefile xmlrpc-c-1.06.04-cflags/lib/expat/xmltok/Makefile
--- xmlrpc-c-1.06.04/lib/expat/xmltok/Makefile	2006-09-04 14:47:37.000000000 +0200
+++ xmlrpc-c-1.06.04-cflags/lib/expat/xmltok/Makefile	2006-09-04 14:52:31.000000000 +0200
@@ -9,7 +9,7 @@
 # we started using the Gcc -Wundef option, that generates a warning, so
 # se set it explicitly to 0 here.
 
-CFLAGS = $(CFLAGS_COMMON) -DXML_BYTE_ORDER=0 $(CFLAGS_PERSONAL) $(CADD)
+CFLAGS = $(CFLAGS_COMMON) -DXML_BYTE_ORDER=0 $(CFLAGS_PERSONAL) $(CADD) -fPIC
 
 LIBLDFLAGS = $(LDFLAGS_VERSINFO) -rpath $(LIBINST_DIR) $(LADD)
 
diff -ur xmlrpc-c-1.06.04/lib/libutil/Makefile xmlrpc-c-1.06.04-cflags/lib/libutil/Makefile
--- xmlrpc-c-1.06.04/lib/libutil/Makefile	2006-09-04 14:47:37.000000000 +0200
+++ xmlrpc-c-1.06.04-cflags/lib/libutil/Makefile	2006-09-04 14:52:56.000000000 +0200
@@ -43,7 +43,7 @@
 libxmlrpc-c_util.la: $(LIBXMLRPC_UTIL_OBJS)
 	$(LIBTOOL) --mode=link $(CCLD) -o $@ $(LIBLDFLAGS) $^
 
-CFLAGS = $(CFLAGS_COMMON) $(CFLAGS_PERSONAL) $(CADD)
+CFLAGS = $(CFLAGS_COMMON) $(CFLAGS_PERSONAL) $(CADD) -fPIC
 
 $(LIBXMLRPC_UTIL_OBJS):%.lo:%.c
 	$(LIBTOOL) --mode=compile $(CC) -c $(INCLUDES) $(CFLAGS) $<
diff -ur xmlrpc-c-1.06.04/lib/libwww_transport/Makefile xmlrpc-c-1.06.04-cflags/lib/libwww_transport/Makefile
--- xmlrpc-c-1.06.04/lib/libwww_transport/Makefile	2005-09-19 03:20:14.000000000 +0200
+++ xmlrpc-c-1.06.04-cflags/lib/libwww_transport/Makefile	2006-09-04 14:53:04.000000000 +0200
@@ -6,7 +6,7 @@
 
 LIBWWW_INCLUDES := $(shell libwww-config --cflags)
 
-CFLAGS = $(CFLAGS_COMMON) $(CFLAGS_PERSONAL) $(CADD)
+CFLAGS = $(CFLAGS_COMMON) $(CFLAGS_PERSONAL) $(CADD) -fPIC
 LDFLAGS = $(LADD)
 
 INCLUDES = -I$(SRCDIR) -I$(SRCDIR)/include -I$(SRCDIR)/lib/util/include \
diff -ur xmlrpc-c-1.06.04/lib/util/Makefile xmlrpc-c-1.06.04-cflags/lib/util/Makefile
--- xmlrpc-c-1.06.04/lib/util/Makefile	2006-09-04 14:47:37.000000000 +0200
+++ xmlrpc-c-1.06.04-cflags/lib/util/Makefile	2006-09-04 14:53:24.000000000 +0200
@@ -34,7 +34,7 @@
 
 INCLUDES = -Iinclude -I$(BUILDDIR)
 
-CFLAGS = $(CFLAGS_COMMON) $(INCLUDES) $(CFLAGS_PERSONAL) $(CADD)
+CFLAGS = $(CFLAGS_COMMON) $(INCLUDES) $(CFLAGS_PERSONAL) $(CADD) -fPIC
 
 %.lo:%.c
 	$(LIBTOOL) --mode=compile $(CC) -c $(CFLAGS) $<
diff -ur xmlrpc-c-1.06.04/lib/wininet_transport/Makefile xmlrpc-c-1.06.04-cflags/lib/wininet_transport/Makefile
--- xmlrpc-c-1.06.04/lib/wininet_transport/Makefile	2006-01-29 04:59:08.000000000 +0100
+++ xmlrpc-c-1.06.04-cflags/lib/wininet_transport/Makefile	2006-09-04 14:53:30.000000000 +0200
@@ -4,7 +4,7 @@
 
 include $(SRCDIR)/Makefile.config
 
-CFLAGS = $(CFLAGS_COMMON) $(CFLAGS_PERSONAL) $(CADD)
+CFLAGS = $(CFLAGS_COMMON) $(CFLAGS_PERSONAL) $(CADD) -fPIC
 LDFLAGS = -lpthread $(LADD)
 
 INCLUDES = -I$(SRCDIR) -I$(SRCDIR)/src
diff -ur xmlrpc-c-1.06.04/src/Makefile xmlrpc-c-1.06.04-cflags/src/Makefile
--- xmlrpc-c-1.06.04/src/Makefile	2006-09-04 14:47:37.000000000 +0200
+++ xmlrpc-c-1.06.04-cflags/src/Makefile	2006-09-04 14:54:16.000000000 +0200
@@ -159,7 +159,7 @@
 	  $(LIBXMLRPC_SERVER_CGI_OBJS) $(LIBXMLRPC_SERVER_CGI_LIBDEP)
 
 
-CFLAGS = $(CFLAGS_COMMON) $(CFLAGS_PERSONAL) $(CADD)
+CFLAGS = $(CFLAGS_COMMON) $(CFLAGS_PERSONAL) $(CADD) -fPIC
 
 $(LIBXMLRPC_CLIENT_OBJS) $(LIBXMLRPC_SERVER_OBJS):%.lo:%.c
 	$(LIBTOOL) --mode=compile $(CC) -c $(INCLUDES) $(CFLAGS) $<
diff -ur xmlrpc-c-1.06.04/src/cpp/Makefile xmlrpc-c-1.06.04-cflags/src/cpp/Makefile
--- xmlrpc-c-1.06.04/src/cpp/Makefile	2006-09-04 14:47:37.000000000 +0200
+++ xmlrpc-c-1.06.04-cflags/src/cpp/Makefile	2006-09-04 14:53:53.000000000 +0200
@@ -136,7 +136,7 @@
 #  RULES TO COMPILE OBJECT MODULES FOR LIBRARIES
 #-----------------------------------------------------------------------------
 
-CXXFLAGS = $(CXXFLAGS_COMMON) $(CFLAGS_PERSONAL) $(CADD)
+CXXFLAGS = $(CXXFLAGS_COMMON) $(CFLAGS_PERSONAL) $(CADD) -fPIC
 
 XmlRpcCpp.o:%.o:%.cpp
 	$(CXX) -c $(BASIC_INCLUDES) $(CXXFLAGS) $<
