diff -ur xmlrpc-c-1.03.10/Makefile.common xmlrpc-c-1.03.10.soname/Makefile.common
--- xmlrpc-c-1.03.10/Makefile.common	2005-06-06 02:27:10.000000000 +0200
+++ xmlrpc-c-1.03.10.soname/Makefile.common	2006-01-11 16:44:45.984603250 +0100
@@ -70,11 +70,11 @@
 $(BUILDDIR)/lib/util/casprintf.lo: FORCE
 	$(MAKE) -C $(dir $@) -f $(SRCDIR)/lib/util/Makefile $(notdir $@) 
 
-$(BUILDDIR)/lib/expat/xmlparse/libxmlrpc_xmlparse.la: FORCE
+$(BUILDDIR)/lib/expat/xmlparse/libxmlrpc-c_xmlparse.la: FORCE
 	$(MAKE) -C $(dir $@) -f $(SRCDIR)/lib/expat/xmlparse/Makefile \
 	    $(notdir $@)
 
-$(BUILDDIR)/lib/expat/xmltok/libxmlrpc_xmltok.la: FORCE
+$(BUILDDIR)/lib/expat/xmltok/libxmlrpc-c_xmltok.la: FORCE
 	$(MAKE) -C $(dir $@) -f $(SRCDIR)/lib/expat/xmltok/Makefile \
 	    $(notdir $@)
 
@@ -90,15 +90,15 @@
 	$(MAKE) -C $(dir $@) -f $(SRCDIR)/lib/libwww_transport/Makefile \
 	    $(notdir $@)
 
-$(BUILDDIR)/src/libxmlrpc.la: FORCE
+$(BUILDDIR)/src/libxmlrpc-c.la: FORCE
 	$(MAKE) -C $(dir $@) -f $(SRCDIR)/src/Makefile \
 	    $(notdir $@)
 
-$(BUILDDIR)/src/libxmlrpc_client.la: FORCE
+$(BUILDDIR)/src/libxmlrpc-c_client.la: FORCE
 	$(MAKE) -C $(dir $@) -f $(SRCDIR)/src/Makefile \
 	    $(notdir $@)
 
-$(BUILDDIR)/src/libxmlrpc_server.la: FORCE
+$(BUILDDIR)/src/libxmlrpc-c_server.la: FORCE
 	$(MAKE) -C $(dir $@) -f $(SRCDIR)/src/Makefile \
 	    $(notdir $@)
 
Only in xmlrpc-c-1.03.10.soname: Makefile.common~
diff -ur xmlrpc-c-1.03.10/configure.in xmlrpc-c-1.03.10.soname/configure.in
--- xmlrpc-c-1.03.10/configure.in	2006-01-11 16:54:42.321872000 +0100
+++ xmlrpc-c-1.03.10.soname/configure.in	2006-01-11 16:45:17.102548000 +0100
@@ -103,7 +103,7 @@
 
 dnl Set up the appropriate Makefile substitutions.
 
-LIBXMLRPC_CLIENT_LA=libxmlrpc_client.la
+LIBXMLRPC_CLIENT_LA=libxmlrpc-c_client.la
 AC_SUBST(LIBXMLRPC_CLIENT_LA)
 CLIENTTEST=clienttest
 AC_SUBST(CLIENTTEST)
@@ -149,7 +149,7 @@
 if test x"$enable_abyss_server" != xno; then
     FEATURE_LIST="abyss-server $FEATURE_LIST"
     ABYSS_SUBDIR=abyss
-    LIBXMLRPC_ABYSS_SERVER_LA=libxmlrpc_abyss_server.la
+    LIBXMLRPC_ABYSS_SERVER_LA=libxmlrpc-c_abyss_server.la
     SERVERTEST=servertest
     VALIDATEE=validatee
     XMLRPC_ABYSS_H=xmlrpc_abyss.h
@@ -189,7 +189,7 @@
 INTEROP_CLIENT_SUBDIR=
 if test x"$enable_cplusplus" != xno; then
     FEATURE_LIST="c++ $FEATURE_LIST"
-    LIBXMLRPC_CPP_A=libxmlrpc_cpp.a
+    LIBXMLRPC_CPP_A=libxmlrpc-c_cpp.a
     CPPTEST=cpptest
     XMLRPCCPP_H=XmlRpcCpp.h
 
@@ -604,7 +604,7 @@
 
     EXTRA_XML_LIBS="$LIBXML2_LIBS"
 else
-    EXTRA_XML_LIBS="-lxmlrpc_xmlparse -lxmlrpc_xmltok"
+    EXTRA_XML_LIBS="-lxmlrpc-c_xmlparse -lxmlrpc-c_xmltok"
 fi
 
 AC_SUBST(EXTRA_XML_LIBS)
Only in xmlrpc-c-1.03.10.soname: configure.in~
diff -ur xmlrpc-c-1.03.10/lib/abyss/src/Makefile xmlrpc-c-1.03.10.soname/lib/abyss/src/Makefile
--- xmlrpc-c-1.03.10/lib/abyss/src/Makefile	2005-08-04 05:31:54.000000000 +0200
+++ xmlrpc-c-1.03.10.soname/lib/abyss/src/Makefile	2006-01-11 16:45:51.308685750 +0100
@@ -20,12 +20,12 @@
 
 INCLUDES = -I$(SRCDIR) -I$(SRCDIR)/include -I$(SRCDIR)/lib/util/include
 
-LTLIBRARIES_TO_INSTALL = libxmlrpc_abyss.la
+LTLIBRARIES_TO_INSTALL = libxmlrpc-c_abyss.la
 
 default: all
 
 .PHONY: all
-all: libxmlrpc_abyss.la
+all: libxmlrpc-c_abyss.la
 
 
 ABYSS_OBJS = \
@@ -41,7 +41,7 @@
   conn.lo \
 
 
-libxmlrpc_abyss.la: $(ABYSS_OBJS)
+libxmlrpc-c_abyss.la: $(ABYSS_OBJS)
 	$(LIBTOOL) --mode=link $(CCLD) -o $@ $(LIBLDFLAGS) $^
 
 $(ABYSS_OBJS):%.lo:%.c
Only in xmlrpc-c-1.03.10.soname/lib/abyss/src: Makefile~
diff -ur xmlrpc-c-1.03.10/lib/expat/xmlparse/Makefile.in xmlrpc-c-1.03.10.soname/lib/expat/xmlparse/Makefile.in
--- xmlrpc-c-1.03.10/lib/expat/xmlparse/Makefile.in	2004-10-16 23:41:28.000000000 +0200
+++ xmlrpc-c-1.03.10.soname/lib/expat/xmlparse/Makefile.in	2006-01-11 16:46:53.592578250 +0100
@@ -106,10 +106,10 @@
 
 INCLUDES = -I$(srcdir)/../xmltok
 
-lib_LTLIBRARIES = libxmlrpc_xmlparse.la
+lib_LTLIBRARIES = libxmlrpc-c_xmlparse.la
 
-libxmlrpc_xmlparse_la_SOURCES = xmlparse.c xmlparse.h
-libxmlrpc_xmlparse_la_LDFLAGS = @VERSION_INFO@
+libxmlrpc_c_xmlparse_la_SOURCES = xmlparse.c xmlparse.h
+libxmlrpc_c_xmlparse_la_LDFLAGS = @VERSION_INFO@
 
 EXTRA_DIST = xmlparse.dsp
 mkinstalldirs = $(SHELL) $(top_srcdir)/mkinstalldirs
@@ -122,8 +122,8 @@
 CPPFLAGS = @CPPFLAGS@
 LDFLAGS = @LDFLAGS@
 LIBS = @LIBS@
-libxmlrpc_xmlparse_la_LIBADD = 
-libxmlrpc_xmlparse_la_OBJECTS =  xmlparse.lo
+libxmlrpc_c_xmlparse_la_LIBADD = 
+libxmlrpc_c_xmlparse_la_OBJECTS =  xmlparse.lo
 CFLAGS = @CFLAGS@
 COMPILE = $(CC) $(DEFS) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS)
 LTCOMPILE = $(LIBTOOL) --mode=compile $(CC) $(DEFS) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS)
@@ -136,8 +136,8 @@
 
 TAR = gtar
 GZIP_ENV = --best
-SOURCES = $(libxmlrpc_xmlparse_la_SOURCES)
-OBJECTS = $(libxmlrpc_xmlparse_la_OBJECTS)
+SOURCES = $(libxmlrpc_c_xmlparse_la_SOURCES)
+OBJECTS = $(libxmlrpc_c_xmlparse_la_OBJECTS)
 
 all: all-redirect
 .SUFFIXES:
@@ -211,8 +211,8 @@
 
 maintainer-clean-libtool:
 
-libxmlrpc_xmlparse.la: $(libxmlrpc_xmlparse_la_OBJECTS) $(libxmlrpc_xmlparse_la_DEPENDENCIES)
-	$(LINK) -rpath $(libdir) $(libxmlrpc_xmlparse_la_LDFLAGS) $(libxmlrpc_xmlparse_la_OBJECTS) $(libxmlrpc_xmlparse_la_LIBADD) $(LIBS)
+libxmlrpc-c_xmlparse.la: $(libxmlrpc_c_xmlparse_la_OBJECTS) $(libxmlrpc_c_xmlparse_la_DEPENDENCIES)
+	$(LINK) -rpath $(libdir) $(libxmlrpc_c_xmlparse_la_LDFLAGS) $(libxmlrpc_c_xmlparse_la_OBJECTS) $(libxmlrpc_c_xmlparse_la_LIBADD) $(LIBS)
 
 tags: TAGS
 
Only in xmlrpc-c-1.03.10.soname/lib/expat/xmlparse: Makefile.in~
diff -ur xmlrpc-c-1.03.10/lib/expat/xmltok/Makefile xmlrpc-c-1.03.10.soname/lib/expat/xmltok/Makefile
--- xmlrpc-c-1.03.10/lib/expat/xmltok/Makefile	2005-05-05 23:37:43.000000000 +0200
+++ xmlrpc-c-1.03.10.soname/lib/expat/xmltok/Makefile	2006-01-11 16:47:09.233555750 +0100
@@ -20,7 +20,7 @@
 include $(SRCDIR)/Makefile.common
 
 .PHONY: all
-all: libxmlrpc_xmltok.la
+all: libxmlrpc-c_xmltok.la
 
 .PHONY: clean
 clean: clean-common
@@ -41,11 +41,11 @@
 .PHONY: dep
 dep: dep-common
 
-LTLIBRARIES_TO_INSTALL = libxmlrpc_xmltok.la
+LTLIBRARIES_TO_INSTALL = libxmlrpc-c_xmltok.la
 
 LIBXMLRPC_XMLTOK_OBJS = xmltok.lo xmlrole.lo
 
-libxmlrpc_xmltok.la: $(LIBXMLRPC_XMLTOK_OBJS)
+libxmlrpc-c_xmltok.la: $(LIBXMLRPC_XMLTOK_OBJS)
 	$(LIBTOOL) --mode=link $(CCLD) -o $@ $(LIBLDFLAGS) $^
 
 $(LIBXMLRPC_XMLTOK_OBJS):%.lo:%.c
Only in xmlrpc-c-1.03.10.soname/lib/expat/xmltok: Makefile~
diff -ur xmlrpc-c-1.03.10/src/Makefile xmlrpc-c-1.03.10.soname/src/Makefile
--- xmlrpc-c-1.03.10/src/Makefile	2005-07-01 05:13:03.000000000 +0200
+++ xmlrpc-c-1.03.10.soname/src/Makefile	2006-01-11 16:56:50.565886750 +0100
@@ -37,8 +37,8 @@
   LIBXML = $(LIBXML2_LIBS)
 else
   LIBXML_INCLUDES = -I$(SRCDIR)/lib/expat/xmlparse
-  LIBXML = $(BUILDDIR)/lib/expat/xmlparse/libxmlrpc_xmlparse.la \
-           $(BUILDDIR)/lib/expat/xmltok/libxmlrpc_xmltok.la
+  LIBXML = $(BUILDDIR)/lib/expat/xmlparse/libxmlrpc-c_xmlparse.la \
+           $(BUILDDIR)/lib/expat/xmltok/libxmlrpc-c_xmltok.la
 endif
 
 ABYSS_INCLUDES = -I$(SRCDIR)/lib/abyss/src
@@ -72,15 +72,15 @@
 	   $(TRANSPORT_INCLUDES) \
 	   $(LIBXML_INCLUDES) \
 
-TARGET_LTLIBRARIES = libxmlrpc.la libxmlrpc_server.la \
-     libxmlrpc_server_abyss.la
+TARGET_LTLIBRARIES = libxmlrpc-c.la libxmlrpc-c_server.la \
+     libxmlrpc-c_server_abyss.la
 
 ifeq ($(MUST_BUILD_CLIENT),yes)
-  TARGET_LTLIBRARIES += libxmlrpc_client.la
+  TARGET_LTLIBRARIES += libxmlrpc-c_client.la
 endif
 
 ifeq ($(ENABLE_CGI_SERVER),yes)
-  TARGET_LTLIBRARIES += libxmlrpc_server_cgi.la
+  TARGET_LTLIBRARIES += libxmlrpc-c_server_cgi.la
 endif
 
 all: $(TARGET_LTLIBRARIES) $(SUBDIRS:%=%/all)
@@ -89,19 +89,19 @@
 
 LIBLDFLAGS = $(LDFLAGS_VERSINFO) -rpath $(LIBINST_DIR) $(LADD)
 
-libxmlrpc.la: $(LIBXMLRPC_OBJS)
+libxmlrpc-c.la: $(LIBXMLRPC_OBJS)
 	$(LIBTOOL) --mode=link $(CCLD) -o $@ $(LIBLDFLAGS) $^
 
-libxmlrpc_client.la: xmlrpc_client.lo $(TRANSPORT_OBJS) $(CASPRINTF_OBJ)
+libxmlrpc-c_client.la: xmlrpc_client.lo $(TRANSPORT_OBJS) $(CASPRINTF_OBJ)
 	$(LIBTOOL) --mode=link $(CCLD) -o $@ $(LIBLDFLAGS) $^
 
-libxmlrpc_server.la: xmlrpc_registry.lo
+libxmlrpc-c_server.la: xmlrpc_registry.lo
 	$(LIBTOOL) --mode=link $(CCLD) -o $@ $(LIBLDFLAGS) $^
 
-libxmlrpc_server_abyss.la: xmlrpc_server_abyss.lo
+libxmlrpc-c_server_abyss.la: xmlrpc_server_abyss.lo
 	$(LIBTOOL) --mode=link $(CCLD) -o $@ $(LIBLDFLAGS) $^
 
-libxmlrpc_server_cgi.la: xmlrpc_server_cgi.lo
+libxmlrpc-c_server_cgi.la: xmlrpc_server_cgi.lo
 	$(LIBTOOL) --mode=link $(CCLD) -o $@ $(LIBLDFLAGS) $^
 
 CFLAGS = $(CFLAGS_COMMON) $(CFLAGS_PERSONAL) $(CADD)
Only in xmlrpc-c-1.03.10.soname/src: Makefile~
diff -ur xmlrpc-c-1.03.10/src/cpp/Makefile xmlrpc-c-1.03.10.soname/src/cpp/Makefile
--- xmlrpc-c-1.03.10/src/cpp/Makefile	2005-11-23 04:13:11.000000000 +0100
+++ xmlrpc-c-1.03.10.soname/src/cpp/Makefile	2006-01-11 16:55:43.925722000 +0100
@@ -18,8 +18,8 @@
   LIBXML = $(LIBXML2_LIBS)
 else
   LIBXML_INCLUDES = -I$(SRCDIR)/lib/expat/xmlparse
-  LIBXML = $(BUILDDIR)/lib/expat/xmlparse/libxmlrpc_xmlparse.la \
-           $(BUILDDIR)/lib/expat/xmltok/libxmlrpc_xmltok.la
+  LIBXML = $(BUILDDIR)/lib/expat/xmlparse/libxmlrpc-c_xmlparse.la \
+           $(BUILDDIR)/lib/expat/xmltok/libxmlrpc-c_xmltok.la
 endif
 
 ifeq ($(ENABLE_LIBXML2),yes)
@@ -28,9 +28,9 @@
   XMLRPC_XML_PARSER = xmlrpc_expat.lo
 endif
 
-LIBXMLRPC = $(BUILDDIR)/src/libxmlrpc.la
-LIBXMLRPC_SERVER = $(BUILDDIR)/src/libxmlrpc_server.la
-LIBXMLRPC_CLIENT = $(BUILDDIR)/src/libxmlrpc_client.la
+LIBXMLRPC = $(BUILDDIR)/src/libxmlrpc-c.la
+LIBXMLRPC_SERVER = $(BUILDDIR)/src/libxmlrpc-c_server.la
+LIBXMLRPC_CLIENT = $(BUILDDIR)/src/libxmlrpc-c_client.la
 
 LIBXMLRPCPP_OBJS = girmem.o value.o fault.o outcome.o param_list.o xml.o
 LIBXMLRPC_SERVERPP_OBJS = registry.o
@@ -62,15 +62,15 @@
 # it obfuscates the build process), I haven't figured out yet how to
 # get the shared libraries to build for all the platforms.
 
-# libxmlrpc_cpp.a is the legacy C++ wrapper library.  The others are the
+# libxmlrpc-c_cpp.a is the legacy C++ wrapper library.  The others are the
 # more elaborate replacements.
 
 TARGET_LIBRARIES = \
-  libxmlrpc_cpp.a \
-  libxmlrpc++.a \
-  libxmlrpc_server++.a \
-  libxmlrpc_server_abyss++.a \
-  libxmlrpc_client++.a \
+  libxmlrpc-c_cpp.a \
+  libxmlrpc-c++.a \
+  libxmlrpc-c_server++.a \
+  libxmlrpc-c_server_abyss++.a \
+  libxmlrpc-c_client++.a \
 
 TARGET_PROGS = cpptest
 
@@ -85,34 +85,34 @@
 
 LDFLAGS += $(LADD)
 
-cpptest:%:%.o libxmlrpc_server_abyss++.a libxmlrpc_server++.a \
-  libxmlrpc_client++.a libxmlrpc++.a libxmlrpc_cpp.a \
+cpptest:%:%.o libxmlrpc-c_server_abyss++.a libxmlrpc-c_server++.a \
+  libxmlrpc-c_client++.a libxmlrpc-c++.a libxmlrpc-c_cpp.a \
   $(LIBXMLRPC) $(LIBXMLRPC_SERVER) $(LIBXMLRPC_CLIENT) $(LIBXML)
 	$(LIBTOOL) --mode=link $(CXXLD) -o $@ $(LDFLAGS) $^
 
 LIBLDFLAGS = $(LADD)
 
-libxmlrpc_cpp.a: XmlRpcCpp.o
+libxmlrpc-c_cpp.a: XmlRpcCpp.o
 	-rm -f $@
 	$(AR) cru $@ $^
 	$(RANLIB) $@
 
-libxmlrpc++.a: $(LIBXMLRPCPP_OBJS)
+libxmlrpc-c++.a: $(LIBXMLRPCPP_OBJS)
 	-rm -f $@
 	$(AR) cru $@ $^
 	$(RANLIB) $@
 
-libxmlrpc_server++.a: $(LIBXMLRPC_SERVERPP_OBJS)
+libxmlrpc-c_server++.a: $(LIBXMLRPC_SERVERPP_OBJS)
 	-rm -f $@
 	$(AR) cru $@ $^
 	$(RANLIB) $@
 
-libxmlrpc_server_abyss++.a: $(LIBXMLRPC_SERVER_ABYSSPP_OBJS)
+libxmlrpc-c_server_abyss++.a: $(LIBXMLRPC_SERVER_ABYSSPP_OBJS)
 	-rm -f $@
 	$(AR) cru $@ $^
 	$(RANLIB) $@
 
-libxmlrpc_client++.a: $(LIBXMLRPC_CLIENTPP_OBJS)
+libxmlrpc-c_client++.a: $(LIBXMLRPC_CLIENTPP_OBJS)
 	-rm -f $@
 	$(AR) cru $@ $^
 	$(RANLIB) $@
Only in xmlrpc-c-1.03.10.soname/src/cpp: Makefile~
diff -ur xmlrpc-c-1.03.10/tools/Makefile.common xmlrpc-c-1.03.10.soname/tools/Makefile.common
--- xmlrpc-c-1.03.10/tools/Makefile.common	2005-10-06 18:15:38.000000000 +0200
+++ xmlrpc-c-1.03.10.soname/tools/Makefile.common	2006-01-11 16:57:25.844091500 +0100
@@ -11,8 +11,8 @@
   CLIENT_LDFLAGS += $(shell wininet-config --libs)
 endif
 
-LIBEXPAT = $(BUILDDIR)/lib/expat/xmlparse/libxmlrpc_xmlparse.la \
-           $(BUILDDIR)/lib/expat/xmltok/libxmlrpc_xmltok.la
+LIBEXPAT = $(BUILDDIR)/lib/expat/xmlparse/libxmlrpc-c_xmlparse.la \
+           $(BUILDDIR)/lib/expat/xmltok/libxmlrpc-c_xmltok.la
 
 ifeq ($(ENABLE_LIBXML2_BACKEND),yes)
   LIBXML = $(LIBXML2_LIBS)
@@ -20,10 +20,10 @@
   LIBXML = $(LIBEXPAT)
 endif
 
-LIBXMLRPC = $(BUILDDIR)/src/libxmlrpc.la
-LIBXMLRPC_CPP = $(BUILDDIR)/src/cpp/libxmlrpc_cpp.a
-LIBXMLRPC_CLIENT = $(BUILDDIR)/src/libxmlrpc_client.la
-LIBXMLRPC_SERVER = $(BUILDDIR)/src/libxmlrpc_server.la
+LIBXMLRPC = $(BUILDDIR)/src/libxmlrpc-c.la
+LIBXMLRPC_CPP = $(BUILDDIR)/src/cpp/libxmlrpc-c_cpp.a
+LIBXMLRPC_CLIENT = $(BUILDDIR)/src/libxmlrpc-c_client.la
+LIBXMLRPC_SERVER = $(BUILDDIR)/src/libxmlrpc-c_server.la
 
 XMLRPC_LIBS = $(LIBXMLRPC) $(LIBXMLRPC_CPP) $(LIBXMLRPC_CLIENT) \
 	$(LIBXMLRPC_SERVER) $(LIBXML)
Only in xmlrpc-c-1.03.10.soname/tools: Makefile.common~
diff -ur xmlrpc-c-1.03.10/xmlrpc-c-config.in xmlrpc-c-1.03.10.soname/xmlrpc-c-config.in
--- xmlrpc-c-1.03.10/xmlrpc-c-config.in	2005-08-04 05:31:52.000000000 +0200
+++ xmlrpc-c-1.03.10.soname/xmlrpc-c-config.in	2006-01-11 16:50:17.989352250 +0100
@@ -54,13 +54,13 @@
 if test "${ENABLE_LIBXML2_BACKEND}" = "yes"; then
   LIBXML="@LIBXML2_LIBS@"
 else
-  LIBXML="-lxmlrpc_xmlparse -lxmlrpc_xmltok"
+  LIBXML="-lxmlrpc-c_xmlparse -lxmlrpc-c_xmltok"
 fi
 
 needCpp=no
 
 the_libdirs="-L@libdir@"
-the_libs="-lxmlrpc ${LIBXML}"
+the_libs="-lxmlrpc-c ${LIBXML}"
 the_rpath=
 the_wl_rpath=
 cpp_libs=
@@ -68,38 +68,38 @@
 while test $# -gt 0; do
   case $1 in
     c++)
-      the_libs="-lxmlrpc_cpp $the_libs"
+      the_libs="-lxmlrpc-c_cpp $the_libs"
 
       # Unfortunately, there is just one legacy CPP library for
       # everything, and it needs all the C libraries -- base, client,
       # and server.  So all legacy C++ programs get linked with client
       # and server libraries, whether they need them or not.
 
-      the_libs="-lxmlrpc_server_abyss $the_libs"
-      the_libs="-lxmlrpc_server $the_libs"
-      the_libs="-lxmlrpc_client $the_libs"
+      the_libs="-lxmlrpc-c_server_abyss $the_libs"
+      the_libs="-lxmlrpc-c_server $the_libs"
+      the_libs="-lxmlrpc-c_client $the_libs"
       ;;
     c++2)
       needCpp=yes
       the_libs="-lxmlrpc++ $the_libs"
       ;;
     cgi-server)
-      the_libs="-lxmlrpc_server $the_libs"
-      the_libs="-lxmlrpc_server_cgi $the_libs"
+      the_libs="-lxmlrpc-c_server $the_libs"
+      the_libs="-lxmlrpc-c_server_cgi $the_libs"
       ;;
     abyss-server)
       the_libs="@LIBABYSS_LDADD@ $the_libs"
-      the_libs="-lxmlrpc_abyss $the_libs"
-      the_libs="-lxmlrpc_server $the_libs"
-      the_libs="-lxmlrpc_server_abyss $the_libs"
+      the_libs="-lxmlrpc-c_abyss $the_libs"
+      the_libs="-lxmlrpc-c_server $the_libs"
+      the_libs="-lxmlrpc-c_server_abyss $the_libs"
       if test "${needCpp}" = "yes"; then
-        the_libs="-lxmlrpc_server++ $the_libs"
-        the_libs="-lxmlrpc_server_abyss++ $the_libs"
+        the_libs="-lxmlrpc-c_server++ $the_libs"
+        the_libs="-lxmlrpc-c_server_abyss++ $the_libs"
         fi
       ;;
     client|libwww-client)
       # libwww-client is for backward compatibility
-      the_libs="-lxmlrpc_client $the_libs"
+      the_libs="-lxmlrpc-c_client $the_libs"
 
       if test "${MUST_BUILD_WININET_CLIENT}" = "yes"; then
         the_libs="@WININET_LDADD@ $the_libs"
@@ -116,7 +116,7 @@
         the_wl_rpath="@LIBWWW_WL_RPATH@ $the_wl_rpath"
       fi
       if test "${needCpp}" = "yes"; then
-        the_libs="-lxmlrpc_client++ $the_libs"
+        the_libs="-lxmlrpc-c_client++ $the_libs"
         fi
       ;;
     --version)
Only in xmlrpc-c-1.03.10.soname: xmlrpc-c-config.in~
diff -ur xmlrpc-c-1.03.10/xmlrpc-c-config.test.in xmlrpc-c-1.03.10.soname/xmlrpc-c-config.test.in
--- xmlrpc-c-1.03.10/xmlrpc-c-config.test.in	2005-06-12 02:06:11.000000000 +0200
+++ xmlrpc-c-1.03.10.soname/xmlrpc-c-config.test.in	2006-01-11 16:57:11.211177000 +0100
@@ -26,15 +26,15 @@
 if test "${ENABLE_LIBXML2_BACKEND}" = "yes"; then
   LIBXML="@LIBXML2_LIBS@"
 else
-  LIBXML="${BUILDDIR}/lib/expat/xmlparse/.libs/libxmlrpc_xmlparse.a"
+  LIBXML="${BUILDDIR}/lib/expat/xmlparse/.libs/libxmlrpc-c_xmlparse.a"
   sopath="${BUILDDIR}/lib/expat/xmlparse/.libs:$sopath"
-  LIBXML="${LIBXML} ${BUILDDIR}/lib/expat/xmltok/.libs/libxmlrpc_xmltok.a"
+  LIBXML="${LIBXML} ${BUILDDIR}/lib/expat/xmltok/.libs/libxmlrpc-c_xmltok.a"
   sopath="${BUILDDIR}/lib/expat/xmltok/.libs:$sopath"
 fi
 
 needCpp=no
 
-the_libs="${BUILDDIR}/src/.libs/libxmlrpc.a ${LIBXML}"
+the_libs="${BUILDDIR}/src/.libs/libxmlrpc-c.a ${LIBXML}"
 the_includes="-I${SRCDIR}/include"
 sopath="${BUILDDIR}/src/.libs:$sopath"
 
@@ -46,33 +46,33 @@
       # and server.  So all legacy C++ programs get linked with client
       # and server libraries, whether they need them or not.
 
-      the_libs="${BUILDDIR}/src/.libs/libxmlrpc_server.a $the_libs"
-      the_libs="${BUILDDIR}/src/.libs/libxmlrpc_server_abyss.a $the_libs"
-      the_libs="${BUILDDIR}/src/.libs/libxmlrpc_client.a $the_libs"
-      the_libs="${BUILDDIR}/src/cpp/libxmlrpc_cpp.a $the_libs"
+      the_libs="${BUILDDIR}/src/.libs/libxmlrpc-c_server.a $the_libs"
+      the_libs="${BUILDDIR}/src/.libs/libxmlrpc-c_server_abyss.a $the_libs"
+      the_libs="${BUILDDIR}/src/.libs/libxmlrpc-c_client.a $the_libs"
+      the_libs="${BUILDDIR}/src/cpp/libxmlrpc-c_cpp.a $the_libs"
       ;;
     c++2)
       needCpp=yes
-      the_libs="${BUILDDIR}/src/cpp/libxmlrpc++.a $the_libs"
+      the_libs="${BUILDDIR}/src/cpp/libxmlrpc-c++.a $the_libs"
       ;;
     cgi-server)
-      the_libs="${BUILDDIR}/src/.libs/libxmlrpc_server.a $the_libs"
-      the_libs="${BUILDDIR}/src/.libs/libxmlrpc_server_cgi.a $the_libs"
+      the_libs="${BUILDDIR}/src/.libs/libxmlrpc-c_server.a $the_libs"
+      the_libs="${BUILDDIR}/src/.libs/libxmlrpc-c_server_cgi.a $the_libs"
       ;;
     abyss-server)
       the_libs="@LIBABYSS_LDADD@ $the_libs"
-      the_libs="${BUILDDIR}/lib/abyss/src/.libs/libxmlrpc_abyss.a $the_libs"
-      the_libs="${BUILDDIR}/src/.libs/libxmlrpc_server.a $the_libs"
-      the_libs="${BUILDDIR}/src/.libs/libxmlrpc_server_abyss.a $the_libs"
+      the_libs="${BUILDDIR}/lib/abyss/src/.libs/libxmlrpc-c_abyss.a $the_libs"
+      the_libs="${BUILDDIR}/src/.libs/libxmlrpc-c_server.a $the_libs"
+      the_libs="${BUILDDIR}/src/.libs/libxmlrpc-c_server_abyss.a $the_libs"
       if test "${needCpp}" = "yes"; then
-        the_libs="${BUILDDIR}/src/cpp/libxmlrpc_server++.a $the_libs"
-        the_libs="${BUILDDIR}/src/cpp/libxmlrpc_server_abyss++.a $the_libs"
+        the_libs="${BUILDDIR}/src/cpp/libxmlrpc-c_server++.a $the_libs"
+        the_libs="${BUILDDIR}/src/cpp/libxmlrpc-c_server_abyss++.a $the_libs"
         fi
       the_includes="-I${SRCDIR}/lib/abyss/src $the_includes"
       sopath="${BUILDDIR}/lib/abyss/src/.libs:$sopath"
       ;;
     client)
-      the_libs="${BUILDDIR}/src/.libs/libxmlrpc_client.a $the_libs"
+      the_libs="${BUILDDIR}/src/.libs/libxmlrpc-c_client.a $the_libs"
       if test "${MUST_BUILD_WININET_CLIENT}" = "yes"; then
         the_libs="@WININET_LDADD@ $the_libs"
         the_rpath="@WININET_RPATH@ $the_rpath"
@@ -89,7 +89,7 @@
         the_wl_rpath="@LIBWWW_WL_RPATH@ $the_wl_rpath"
         fi
       if test "${needCpp}" = "yes"; then
-        the_libs="${BUILDDIR}/src/cpp/libxmlrpc_client++.a $the_libs"
+        the_libs="${BUILDDIR}/src/cpp/libxmlrpc-c_client++.a $the_libs"
         fi
       ;;
     --version)
Only in xmlrpc-c-1.03.10.soname: xmlrpc-c-config.test.in~
